File: cs/algorithms_in/selection_sort.cs
namespace algorithms_in;

public class SelectionSort
{
    public static void selectionSort(int[] array) {
        for (int i = 0; i < array.Length; i++) {
            int currentMin = array[i];
            int currentMinIndex = i;

            for (int j = i + 1; j < array.Length; j++) {
                if (currentMin > array[j]) {
                    currentMin = array[j];
                    currentMinIndex = j;
                }
            }

            if (currentMinIndex != i) {
                array[currentMinIndex] = array[i];
                array[i] = currentMin;
            }
        }
    }
}

File: rust/algorithms_in/src/selection_sort.rs
pub fn selection_sort(array: &mut [usize]) {
    for i in 0..array.len() {
        let mut current_min: usize = array[i];
        let mut current_min_index: usize = i;
        for j in (i + 1)..array.len() {
            if current_min > array[j] {
                current_min = array[j];
                current_min_index = j
            }
        }
        if current_min_index != i {
            array[current_min_index] = array[i];
            array[i] = current_min;
        }
    }
}